email-smtp/src/main/java/io/commercetools/sunrise/email/smtp/SmtpAuthEmailSender.java

```diff
@@ -1,27 +1,242 @@
+     * sender waits to be connected to the SMTP server, waits to read data from the SMTP server, or waits to write data
+     * to the SMTP server, an {@link EmailSenderException} will be raised and sending of the e-mail that caused the
+     * issue will be aborted. If this happens before the e-mail has been received by the SMTP server, the e-mail will
+     * not be sent, as users of this service will not typically implement error-handling in such cases. In general,
+     * shorter timeouts may lead to more messages being lost in the abscence of further error handling, longer timeouts
+     * may lead to more tasks being queued if a {@link ForkJoinPool} is used as {@link Executor}; run-time monitoring of
+     * the {@link ForkJoinPool} might help choose a suitable timeout value.
+     * <h1>Message size</h1>
+     * The e-mail sender does not limit the size of the messages it accepts. Because the {@link #send(MessageEditor)}
+     * method fills messages before sending is attempted within the {@link Executor}, it is theoretically possible that
+     * a large queue of messages waiting to be send consumes all available memory. Users of this API should therefore
+     * ensure that messages do not exceed a certain size.
+     * <h1>Configuration correctness</h1>
+     * Note that this constructor does not fail fast: the constructor does not create a connection to the mail server to
+     * ensure that the connection details and credentials are correct. If you would like to ensure a correct
+     * configuration, send a test message at startup and decide on your own what to do if that fails.
+     * <h1>Shutdown</h1>
+     * If the JVM running this e-mail sender is shut down, the {@link Executor} passed to this constructor should be
+     * shut down, too, in a way that ensures that all messages submitted to {@link #send(MessageEditor)} have been sent
+     * via SMTP - or a timeout expires in the case too many messages are waiting. If a {@link ForkJoinPool} is used as
+     * {@link Executor}, {@link ForkJoinPool#awaitQuiescence(long, TimeUnit)} may be used for this purpose.
+     *
+     * @param executor          the executor to use, e.g. {@link ForkJoinPool#commonPool()} may be used, but see above
+     * @param host              the name of the host running the SMTP server to use
+     * @param port              the port of the SMTP server to use
+     * @param transportSecurity how to secure the SMTP connection
+     * @param username          the username to use for authenticating with the SMTP server
+     * @param password          the password to authenticate with
+     * @param timeoutMs         the timeout for creating, reading from and writing to SMTP connections in
+     *                          milliseconds, see above for details
+     */
+    public SmtpAuthEmailSender(@Nonnull final Executor executor, @Nonnull final String host, final int port,
```

Comment:
I agree that the constructor parameters may be grouped into two concerns. Re-arranging constructor arguments makes good sense, i.e. to have `executor` come after `password` and before `timeoutMs`.

I'd rather not want to modify the constructor to fit anticipated future use cases before they have arrived. I.e. I'd refrain from introducing a builder for the SMTP-related constructor arguments. That would create a second way of modify the configuration besides overriding SMTPAuthEmailSender.properties(...) - before we know which properties/parameters actual users of the API need to add.

I could add a constructor with default values for executor and timeoutMs. I'm personally rather not a fan of providing a fast start because fast solutions typically overlook corner cases and require time investment later on when you though you were done already. But whether or not fast startup shall be supported is a decision that you need to make. If I were to choose defaults, I'd use ForkJoinPool.commonPool() as default executor and a timeout of 1000 ms - the latter one being a mere guess that does not consider any context in which the email sender might be used.